### ISSUE TITLE
Modifying material's materialTypeVersion property to an invalid value does not trigger Asset Processor fail pop up

### ISSUE THREAD

**Issue by LB-ArturZieba** [o3de/o3de#8529]
**Describe the bug**
If the material's materialTypeVersion property is modified to an invalid value (e.g. "a") it does not cause the Asset Processor fail to pop up, even though the material fails processing. It should be noted that changing materialType to an invalid property causes the Asset Processor fail to pop up. It should also be noted that if the material is assigned to an entity it keeps being applied properly.

**Steps to reproduce**
1. Go to Tools > Material Editor.
2. Go to File > New and click OK to save a default material in the default save location.
3. Select the Ground child entity of the Atom Default Environment, and assign the material to its Material component.
4. Find the material saved in step 2 in File Explorer, and open it in a text-editing software.
5. Change the materialTypeVersion property value from "4" to "a".

**Expected behavior**
An Asset Processor fail pop up is displayed, and the material does not display properly on the assigned entity.

**Actual behavior**
There is no Asset Processor fail pop up, material works properly on assigned entities.

**Video**

https://user-images.githubusercontent.com/86952082/160123001-921b8d20-0a08-4cf6-8c06-a6a2874ca917.mp4

**Found in Branch**
Development (1f1f449)

**Desktop**
 - Device: PC
 - OS: Windows
 - Version 10
 - CPU AMD Ryzen 5 3600
 - GPU NVIDIA GeForce RTX 2060 SUPER
 - Memory 16GB

**Comment by santorac** [comment]
This is not a sig-graphics issue. 

There are two separate concerns: 
1) The missing windows notification when the AP fails to process the material in some cases. The difference is when you change the materialType to a file that doesn't exist, the ProcessJob step fails. When you change the materialTypeVersion value to something that's invalid JSON, the CreateJobs step fails. It looks like the underlying issue is that popup notifications do not occur for CreateJobs failures. 
2) The material continues to be displayed in the Editor with no indication of any issues, even though processing failed. This is part of the asset system's design. Because the updated _source_ file has failed, the runtime is continuing to use the old _product_ file in the cache. 

I believe both these concerns are the responsibility of @o3de/sig-core. Reassigning...

**Comment by lawsonamzn** [comment]
Yes, the default behavior for asset processor is to keep old products available until new ones succed - however, builders can actually opt out of this behavior if so desired by setting a flag on the jobs/builder itself.  If that flag is set, jobs that fail will cause the prior products in the cache to be deleted, which will clear the assets entirely, and make them unavailable in the editor.  It depends on the desires of the builder/system writer.

**Comment by santorac** [comment]
Based on the title, I'm going to assume the main concern here is the notification when CreateJobs failed. @LB-ArturZieba if you want to pursue the concern about the old material continuing to be displayed, then please open a separate Issue for that, thanks.

**Comment by LB-ArturZieba** [comment]
Based on the above comments, I agree that the material being displayed after failing can be disregarded as an issue, thanks for the clarification on that! Based on this, the issue can be closed when the CreateJobs problem is solved, as you have described in the comment.

**Comment by amzn-pratikpa** [comment]
Changed to sig/content as this code lives under their code area
